diff --git a/libtar/decode.c b/libtar/decode.c
index b34b1d9..d221c3a 100644
--- a/libtar/decode.c
+++ b/libtar/decode.c
@@ -44,11 +44,6 @@ uid_t
 th_get_uid(const TAR *t)
 {
 	int uid;
-	struct passwd *pw;
-
-	pw = getpwnam(t->th_buf.uname);
-	if (pw != NULL)
-		return pw->pw_uid;
 
 	/* if the password entry doesn't exist */
 	sscanf(t->th_buf.uid, "%o", &uid);
@@ -60,11 +55,6 @@ gid_t
 th_get_gid(const TAR *t)
 {
 	int gid;
-	struct group *gr;
-
-	gr = getgrnam(t->th_buf.gname);
-	if (gr != NULL)
-		return gr->gr_gid;
 
 	/* if the group entry doesn't exist */
 	sscanf(t->th_buf.gid, "%o", &gid);
